etcd/src/main/java/org/jclouds/etcd/features/MembersApi.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jclouds.etcd.features;
+
+import java.util.List;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+
+import org.jclouds.etcd.domain.members.CreateMember;
+import org.jclouds.etcd.domain.members.Member;
+import org.jclouds.rest.annotations.BinderParam;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.SelectJson;
+import org.jclouds.rest.binders.BindToJsonPayload;
+import org.jclouds.etcd.fallbacks.EtcdFallbacks.BooleanOnDeleteNonExistentMemberAnd404;
+import org.jclouds.etcd.fallbacks.EtcdFallbacks.NullOnAddExistingMemberAnd409;
+
+@Consumes(MediaType.APPLICATION_JSON)
+@Path("/{jclouds.api-version}/members")
+public interface MembersApi {
+
+   /**
+    * @return list of members within cluster
+    */
+   @Named("members:list")
+   @SelectJson("members")
+   @GET
+   List<Member> list();
+
+   /**
+    * @param member
+    *           non-existing member to add to cluster
+    * @return newly created member or null if member previously existed. Because
+    *         this call returns before registration is fully complete it is not
+    *         guaranteed that the Member class will have any variable, other than
+    *         peerURLs, from the CreateMember class present. 
+    */
+   @Named("members:add")
+   @Fallback(NullOnAddExistingMemberAnd409.class)
```

Comment:
Don't use 409 fallbacks in POST/PUT methods. What does a `null` value mean to users? Although documented it is not a good practice: does it mean the member was not created (and does not exist) because the operation failed? Does it mean there was one that already existed? Returning `null` is too ambiguous and we don't use fallbacks in write operations (POST/PUT).

Let the method throw an exception. jclouds already has the `ResourceAlreadyExistsException` exception, so configure the error handler to throw it in this case instead of using a fallback.